### DISCUSSION TITLE
Respaldo Sistema Comercial / usuario activo (sincronizador)

### DISCUSSION THREAD

**Discussion by jparadaa** [AndresRamos/ARSoftware.Contpaqi.Comercial.Api#33] [General]
Este tema no tiene que ver con el API, pero a ver si podemos aportar algunas ideas...

Aquí en la empresa se hacen dos respaldos al día, un respaldo se realiza regularmente en un horario de 2-3pm y otro respaldo al finalizar labores 6pm, el tema es que tengo que cancelar el sincronizador, porque como sabemos, si está activo no pueden realizar el respaldo.

Qué recomendarías, alguna tarea programada o algo así donde pueda cancelar el sincronizador en un tiempo programado y luego volver a ejecutar?

**Comment by AndresRamos** [comment]
Con la version actual `v2.4.0` puedes configurar el Sincronizador para cerrarse a una cierta hora modificando el archivo `appsettings.json`. Puedes asignar la hora en `ShutdownTime` del objecto `ApiSyncConfig`.

Por ejemplo, puedes decirle al Sincronizador que se termine a las 6pm:
```json
"ApiSyncConfig": {
  "ShutdownTime": "18:00:00"
}
```

Para iniciarlo puedes configurar una tarea con el Task Scheduler de Windows para levantar el Sincronizador en la mañana, pero esto no soluciona tu problema ya que tu debes iniciar y cerrar el Sincronizador dos veces durante el dia.

El Sincronizador es un proceso de consola que se puede cancelar con el comando `CTRL+C`. Podrías averiguar cómo cancelar un proceso de consola con tu lenguaje de programación.

De mi lado se me ocurre que pudiera implementar un `ContpaqiRequest` para cerrar el Sincronizador. Algo como `ShutdownSyncRequest`, y al procesar esa solicitud el Sincronizador se termina y cierra. Tendrías comoquiera que ver la manera de como volverlo a levantar manualmente o con otra tarea programada.

**Comment by jparadaa** [comment]
Hola Andrés, voy a investigar un poco más el tema en cuanto tenga alguna idea de solución comentaré por acá
